### ISSUE TITLE
`PatternVariableAssignment` gives error when assigning to field (using `this.`) with the same name as pattern variable

### ISSUE THREAD

**Issue by james-baker-aera** [checkstyle/checkstyle#18132]
I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/patternvariableassignment.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Example.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="PatternVariableAssignment"/>
    </module>
</module>

/var/tmp $ cat Example.java
class Example {

    String string;

    void setString(Object object) {
        if (object instanceof String string) {
            this.string = string;
        }
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-12.1.2-all.jar -c config.xml Example.java
Starting audit...
[ERROR] /var/tmp/Example.java:7:18: Assignment of pattern variable 'string' is not allowed. [PatternVariableAssignment]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

There should not be any errors. I'm not assigning to the pattern variable `string`. I'm assigning to the field `this.string`.

**Comment by romani** [comment]
@ivamly, can you help us with fix for this problem?

**Comment by ivamly** [comment]
@romani Yes, of course! I'll look into this. Do you have any advice or guidance?

**Comment by romani** [comment]
No , just create test and enjoy debuging
 Test driven development

**Comment by romani** [comment]
Fix is merged
